src/api/ucc.h

```diff
@@ -0,0 +1,709 @@
+    UCC_ATTRIBS_OUTSTANDING_CALLS      = UCC_BIT(1),
+    UCC_ATTRIBS_EP                     = UCC_BIT(2),
+    UCC_ATTRIBS_EP_TYPE                = UCC_BIT(3),
+    UCC_ATTRIBS_SYNC_TYPE              = UCC_BIT(4),
+    UCC_ATTRIBS_OOB                    = UCC_BIT(5),
+    UCC_MEM_PARAMS                     = UCC_BIT(6)
+};
+
+typedef enum  {
+    UCC_MEM_SYMMETRIC   = UCC_BIT(0),
+    UCC_MEM_PERSISTENT  = UCC_BIT(1),
+    UCC_MEM_ALIGN32     = UCC_BIT(2),
+    UCC_MEM_ALIGN64     = UCC_BIT(3),
+    UCC_MEM_ALIGN128    = UCC_BIT(4),
+} ucc_mem_constraints_t;
+
+typedef enum {
+	UCC_REMOTE_ATOMICS,
+	UCC_REMOTE_COUNTERS
+} ucc_mem_hints_t;
+
+typedef struct ucc_mem_map_params {
+    void                    *address;
+    size_t                  len;
+    ucc_mem_hints_t         hints;
+    ucc_mem_constraints_t   constraints;
+} ucc_mem_map_params_t;
+
+typedef struct  ucc_team_oob_coll {
+    int             (*allgather)(void *src_buf, void *recv_buf, size_t size,
+                                 void *allgather_info,  void **request);
+    ucc_status_t    (*req_test)(void *request);
+    ucc_status_t    (*req_free)(void *request);
+    uint32_t 	     participants;
+    void             *coll_info;
+}  ucc_team_oob_coll_t;
+
+typedef enum {
+    UCC_COLLECTIVE_POST_ORDERED = 0,
+    UCC_COLLECTIVE_POST_UNORDERED = 1
+} ucc_post_ordering_t;
+
+typedef enum {
+    UCC_COLLECTIVE_EP_RANGE_CONTIG = 0, 
+    UCC_COLLECTIVE_EP_RANGE_NONCONTIG = 1
+} ucc_ep_range_type_t;
+
+typedef struct ucc_team_params {
+    uint64_t                mask;
+    ucc_post_ordering_t     ordering;
+    uint64_t                outstanding_colls;
+    uint64_t                ep;
+    ucc_ep_range_type_t     ep_range;
+    ucc_coll_sync_type_t    sync_type;
+    ucc_team_oob_coll_t     oob_collective;
+    ucc_mem_map_params_t    mem_params;
+} ucc_team_params_t;
+
+typedef struct ucc_team_attribs {
+    uint64_t               mask;
+    ucc_post_ordering_t    ordering;
+    uint64_t               outstanding_colls;
+    uint64_t               ep;
+    ucc_ep_range_type_t    ep_range;
+    ucc_coll_sync_type_t   sync_type;
+    ucc_mem_map_params_t   mem_params;
+} ucc_team_attribs_t;
+
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @brief ucc_team_create_post is a nonblocking collective operation to create
+ * the team handle. It takes in parameters ucc_context_h, num_handles,
+ * ucc_team_params_t and returns a ucc_team_handle_h. The ucc_team_params_t
+ * provides user configuration to customize the team. The routine returns
+ * immediately after posting the operation with the new team handle. However,
+ * the team handle is not ready for posting the collective operation.
+ * ucc_team_create_test operation is used to learn the status of the new team
+ * handle. On error, the team handle will not be created and corresponding error
+ * code as defined by ucc_status_t is returned.
+ *
+ * @param  [in] contexts     Communication context abstracting the resources
```

Comment:
params description of ucc_team_create_post is not consistent with actual function definition

Reply:
Will fix it.